src/transformations/add-storage-gaps.ts

```diff
@@ -21,54 +21,60 @@ export function* addStorageGaps(
     if (contract.contractKind === 'contract') {
       let targetSlots = DEFAULT_SLOT_COUNT;
       for (const entry of extractNatspec(contract)) {
-        if (entry.title === 'custom' && entry.tag === 'contract-size') {
+        if (entry.title === 'custom' && entry.tag === 'storage-size') {
           targetSlots = parseInt(entry.args);
         }
       }
 
       const contractSize = getContractSize(contract, getLayout(contract));
       const gapSize = Math.floor((32 * targetSlots - contractSize) / 32);
 
-      const contractBounds = getNodeBounds(contract);
-      const start = contractBounds.start + contractBounds.length - 1;
+      if (gapSize > 0) {
+        const contractBounds = getNodeBounds(contract);
+        const start = contractBounds.start + contractBounds.length - 1;
 
-      const text = formatLines(0, [
-        ``,
-        [
-          `/**`,
-          ` * This empty reserved space is put in place to allow future versions to add new`,
-          ` * variables without shifting down storage in the inheritance chain.`,
-          ` * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps`,
-          ` */`,
-          `uint256[${gapSize}] private __gap;`,
-        ],
-      ]);
+        const text = formatLines(0, [
+          ``,
+          [
+            `/**`,
+            ` * This empty reserved space is put in place to allow future versions to add new`,
+            ` * variables without shifting down storage in the inheritance chain.`,
+            ` * See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps`,
+            ` */`,
+            `uint256[${gapSize}] private __gap;`,
+          ],
+        ]);
 
-      yield {
-        kind: 'add-storage-gaps',
-        start,
-        length: 0,
-        text,
-      };
+        yield {
+          kind: 'add-storage-gaps',
+          start,
+          length: 0,
+          text,
+        };
+      } else if (gapSize < 0) {
```

Comment:
Can you make this the first `if` branch? I find it better to see the error branch first.